src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -56,16 +54,34 @@ public bool TryRegister(SafeSocketHandle socket, out Interop.Error error)
 
         private static readonly object s_lock = new object();
 
-        // In debug builds, force there to be 2 engines. In release builds, use half the number of processors when
-        // there are at least 6. The lower bound is to avoid using multiple engines on systems which aren't servers.
-#pragma warning disable CA1802 // const works for debug, but needs to be static readonly for release
-        private static readonly int s_engineCount =
-#if DEBUG
-            2;
-#else
-            Environment.ProcessorCount >= 6 ? Environment.ProcessorCount / 2 : 1;
-#endif
-#pragma warning restore CA1802
+        private static readonly int s_engineCount = GetEnginesCount();
+
+        private static int GetEnginesCount()
+        {
+            if (uint.TryParse(Environment.GetEnvironmentVariable("DOTNET_SYSTEM_NET_SOCKETS_THREAD_COUNT"), out uint count))
+            {
+                return (int)count;
+            }
+
+            // the data that stands behind this heuristic can be found at https://github.com/dotnet/runtime/pull/35800#issuecomment-624719500
+            // the goal is to have a single epoll thread per every 28 cores
+            const int coresPerSingleEpollThread = 28;
```

Comment:
Nit:
```suggestion
            const int CoresPerThread = 28;
```
- Cores => cores
- We use epoll on Linux but kqueues on mac
- Not sure what "single" meant :)

Reply:
`Cores` -> `Processors`